plugin/template/template.go

```diff
@@ -105,7 +105,7 @@ func (h Handler) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg)
 		}
 
 		w.WriteMsg(msg)
-		return template.rcode, nil
+		return dns.RcodeSuccess, nil
```

Comment:
I think this is correct, but it would be counter-intuitive to someone unfamiliar with the code.
So maybe a small comment in the code to describe why this is always set to Success, regardless of the actual response code.  Perhaps something like ...
```
// Template always writes to client, so always return success here to avoid a second write to client for "error" responses.
```

Reply:
Done